include/tvm/runtime/object.h

```diff
@@ -350,6 +350,39 @@ inline RelayRefType GetRef(const ObjectType* ptr);
 template <typename SubRef, typename BaseRef>
 inline SubRef Downcast(BaseRef ref);
 
+template <typename T> class ObjectPtr;
+
+/*!
+ * \brief Get an object ptr type from a raw object ptr.
+ *
+ * \param ptr The object pointer
+ * \tparam BaseType The reference type
+ * \tparam ObjectType The object type
+ * \return The corresponding ObjectPtr type
+ */
+template <typename BaseType, typename ObjectType,
+          typename std::enable_if<!std::is_same<BaseType, ObjectType>::value, int>::type = 0>
+inline ObjectPtr<BaseType> GetObjectPtr(ObjectType* ptr);
+
+/*!
+ * \brief Get an object ptr type from a raw object ptr.
+ *
+ * \param ptr The object pointer
+ * \tparam ObjectType The object type
+ * \return The corresponding ObjectPtr type
+ */
+template <typename ObjectType>
+inline ObjectPtr<ObjectType> GetObjectPtr(ObjectType* ptr);
```

Comment:
Let us simply unify this function with the above one so that users won't be confused about the variants

Reply:
How is that unification possible?  If you say `GetObjectPtr(op)`, without the explicit `BaseType`, the compiler won't be able to deduce it if `GetObjectPtr` is a template with two independent type parameters.

Reply:
That is what I mean, we should  use GetObjectPtr<Object> as it is not a quite frequently used feature